### PR TITLE
Create StonePaperScissors

### PR DESCRIPTION
**Pull request by DevanshPaliwal**
### **Description:**
The Stone-Paper-Scissors game is a classic hand game where two players simultaneously choose one of three options: Stone, Paper, or Scissors. The outcome of the game is determined by the rules: Stone beats Scissors, Scissors beats Paper, and Paper beats Stone. This project involves creating a simple console-based Stone-Paper-Scissors game in Java.

### **Changes Made:**

**User Interface Enhancement:**

Improved the user interface by adding clear instructions and a user-friendly menu for player selection.
Enhanced the readability and clarity of the text-based interface.

**Input Validation:**

Added input validation to ensure that the player's choice is valid (either Stone, Paper, or Scissors).

**Randomized Opponent:**

Implemented an opponent that randomly selects its move (Stone, Paper, or Scissors) for a more engaging and challenging gaming experience.

### **Testing Done:**

**Unit Testing:**

Performed extensive unit testing on individual functions to verify their correctness.

**Integration Testing:**

Tested the integration of various components of the game to ensure they work together seamlessly.
Simulated multiple game sessions to validate the game logic.

**UI Testing:**

Conducted user interface testing to ensure that the user instructions and menus are clear and user-friendly.

### **Additional Notes:**

The game can be easily extended to include more complex game variants or additional features, such as keeping track of the player's score or implementing multiplayer modes.
Future improvements can involve creating a graphical user interface (GUI) for a more visually appealing gaming experience.
Care was taken to make the code readable and maintainable to facilitate potential updates and enhancements.
Proper documentation was provided to make the code accessible to other developers or contributors.
The Stone-Paper-Scissors game in Java offers a fun and educational project for those looking to practice Java programming.

### CODE REVIEW

File: 16_FirstContribution/StonePaperScissors [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import java.util.Scanner;
+import java.util.Random;
+
+class spsgame {
```
Comment (itsyourap) [inline 16_FirstContribution/StonePaperScissors:4]: The class name should be same as the file name.

Also the file name should end with `.java` extension

File: 16_FirstContribution/StonePaperScissors [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        Scanner ex = new Scanner(System.in);
```
Comment (itsyourap) [inline 16_FirstContribution/StonePaperScissors:7]: Defining two scanner objects is unnecessary. Only one scanner object can be used for all inputs

**Review by itsyourap — CHANGES_REQUESTED** [review]

File: 16_FirstContribution/StonePaperScissors (hunk 2 above)
Comment (DevanshPaliwal) [reply 16_FirstContribution/StonePaperScissors:7]: I removed one but it didn't change while submitting

File: 16_FirstContribution/StonePaperScissors (hunk 1 above)
Comment (DevanshPaliwal) [reply 16_FirstContribution/StonePaperScissors:4]: It's not compulsory unless the class is public

**Pull request merged**